### PR TITLE
chore: Adjust global allocator logic switch to be uniform

### PR DESCRIPTION
**Pull request by blt**
We have two places in Vector where we change which global allocator is in use,
`src/lib.rs` and then in a VRL tool. This commit adjusts the logic of both sites to be
uniform.

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

### CODE REVIEW

File: lib/vrl/tests/src/main.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ use vrl::VrlRuntime;
 use vrl::{diagnostic::Formatter, state, Runtime, Terminate, Value};
 use vrl_tests::{docs, Test};
 
-#[cfg(not(target_env = "msvc"))]
+#[cfg(feature = "jemalloc")]
```
Comment (jszwedko) [inline lib/vrl/tests/src/main.rs:20]: I think the semantics here are slightly different now. If I run `cargo` in `lib/vrl/tests` it'll now use the system allocator by default on *nix where before it used `jemalloc`. It's not a big deal, I think we only had this flag in here when we ran into a weird bug with jemalloc that we never quite resolved, just worked around: https://github.com/vectordotdev/vector/issues/6726

Comment (jszwedko) [reply lib/vrl/tests/src/main.rs:20]: Though, maybe it's better to keep the tests running under `jemalloc` by default in case we hit more weird edge cases like that 😅

Comment (blt) [reply lib/vrl/tests/src/main.rs:20]: Ah, right, the semantics are different. The `unix` feature doesn't properly get flagged on in this crate.
